2048Solution/main/java/models/GameBoard.java

```diff
@@ -0,0 +1,70 @@
+package main.java.models;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GameBoard {
+
+
+    private int size;
+    private int[][] board;
+    private int rowToBeFilled;
```

Comment:
These should not be properties of this class. These are temporary variables generally denoting a parameter.

Reply:
size -> denotes size of the board
board[][] -> denotes the elements in board
rowToBeFilled -> denotes the random value for the rowNumber in the board
columnToBeFilled -> denotes the random value for the columnNumber in the board
getRandomFrom -> is an Array list which contains all the empty grid number available in board.
which of these variables shall I remove, and how can I use instead?

Reply:
rowToBeFilled and columnToBeFilled. These are not the properties of a board, right?

What should a GameBoard have? size, board and maybe emptyCells for easier access.

Everything else should be passed as parameters to the function that is using it.

Name properties based on what it holds with respect to the class and not based on how you're going to use it in some function.